### ISSUE TITLE
Some question about evaluate

### ISSUE THREAD

**Issue by zml110120** [XuYunqiu/H2FA_R-CNN#9]
## ❓ Some question about evaluate
I'm using my own data set (only 14 categories), but pred_classes in the red box below is greater than 14. What's the problem? How can I modify it? I have been registered our dataset in ` detectron2/ data/datasets/cdod.py ` .

![image](https://user-images.githubusercontent.com/29172684/209575727-be2f8e08-c9f7-4211-baf6-26f187294821.png)

**Comment by XuYunqiu** [comment]
Could you please provide more details of your datasets?
1. Which datasets are used as the source and target datasets respectively? 
2. How many categories are there in the two datasets?
3. Which pre-trained model is used for evaluation?

**Comment by zml110120** [comment]
@XuYunqiu Dear author,
1. `cp_train cp_val` are my own datasets. I use `cp_train` as train target, `voc_2012_trainval` as train source, and `cp_val` as test. As shown in the first picture.

2. `cp_train cp_val` datasets both have 14 categories.

3. The pre-train model uses the already generated model (trained by `cp_train  voc_2012_trainval`). But during pre-train phase, i had not change `num_classes` in yaml file. As shown in the second picture. I consulted you for advice in other issue https://github.com/XuYunqiu/H2FA_R-CNN/issues/8#issuecomment-1365742212 , but there is still some problems.


![image](https://user-images.githubusercontent.com/29172684/209643741-7a835c74-8afb-46e4-b106-cded325f4ac6.png)


![image](https://user-images.githubusercontent.com/29172684/209644886-d2adc032-d0b1-47fc-ab23-d32d9ce666f4.png)

**Comment by XuYunqiu** [comment]
The categories in both source and target datasets should be the same, in terms of total number and class index.

The source dataset `voc_2012_trainval` is the standard VOC dataset with 20 classes, while the target datasets `cp_train` and `cp_val` have 14 classes. 
During training, the detector is learning to predict 20 classes. Thus, you may get the out-of-index issue.

You can solve this issue by:
1. Besides registering `cp_train` and `cp_val`, you should also register a custom VOC dataset with the same class number and class index with `cp_train` and `cp_val`.
2. Setting `NUM_CLASSES` to 14, in both training and inference stages.

**Comment by zml110120** [comment]
@XuYunqiu 

`cp_train` has `child tree bird monkey ...` classes, but `voc_2012_trainval` has not these classes. After preprocess(the first picture, `cp_train` and `voc_2012_trainval` both use same classes ), can i use `cp_train` as target and `voc_2012_trainval` as source?

![image](https://user-images.githubusercontent.com/29172684/209656202-40ffc12d-7d77-4eb2-b71b-5766d17008cd.png)

**Comment by zml110120** [comment]
@XuYunqiu  Dear author,
I will try your advice!
I still have some questions:
1. Order of `pred_classes` wherether same as `CLASS_NAMES_14`? As shown in first picture.

2. The origin code can save model per 5000 iters. How can i change it to specified iters? Such as save model per 2000 iters?

![image](https://user-images.githubusercontent.com/29172684/209665159-e9cb33ce-886c-4140-a363-163520c65143.png)


![image](https://user-images.githubusercontent.com/29172684/209664961-01159f2d-cf97-44f6-ad1f-230c06b431ee.png)

**Comment by XuYunqiu** [comment]
1. The order of them is the same. But the class index is from 0 to 13, if there are 14 classes. Another thing is that our H<sup>2</sup>FA R-CNN currently only supports CDWSOD which assumes the source and target domains have the **SAME** label space. I'm not sure if it still works well, when the target domain contains some novel classes.

2. You can set `SOLVER.CHECKPOINT_PERIOD` to 2000 in the config file.

**Comment by zml110120** [comment]
Hi,
In your model evaluation phase, the batch size looks like equal 1. How can i change the evaluation phase batch size?

![image](https://user-images.githubusercontent.com/29172684/209985095-42d7519f-244c-48e0-b493-a1cc2ab2a8ce.png)

**Comment by XuYunqiu** [comment]
It's not supported as https://detectron2.readthedocs.io/modules/data.html#detectron2.data.build_detection_test_loader says, so writing some custom code will be needed.

Perhaps, you can do it by hacking the code here
https://github.com/XuYunqiu/H2FA_R-CNN/blob/499c1d90833c475fd9f288bb6bb90829989def64/detectron2/data/build.py#L393-L396
Just change the batch size in the `BatchSampler`.

**Comment by XuYunqiu** [comment]
I'm closing this issue, but let me know if you have further questions.
